tgui/packages/tgui/interfaces/PaperSheet.tsx

```diff
@@ -0,0 +1,881 @@
+    const useBold = held_item_details?.use_bold || false;
+
+    const [inputFieldData, setInputFieldData] = useLocalState(
+      this.context,
+      'inputFieldData',
+      {}
+    );
+
+    const [textAreaText, setTextAreaText] = useLocalState(
+      this.context,
+      'textAreaText',
+      ''
+    );
+
+    const interactMode =
+      held_item_details?.interaction_mode || InteractionType.reading;
+
+    const savableData =
+      textAreaText.length || Object.keys(inputFieldData).length;
+
+    return (
+      <>
+        <PaperSheetStamper scrollableRef={this.scrollableRef} />
+        <Flex direction="column" fillPositionedParent>
+          <Flex.Item grow={3} basis={1}>
+            <PreviewView
+              key={`${raw_field_input?.length || 0}_${
+                raw_text_input?.length || 0
+              }`}
+              scrollableRef={this.scrollableRef}
+              handleOnScroll={this.onScrollHandler}
+              textArea={textAreaText}
+            />
+          </Flex.Item>
+          {interactMode === InteractionType.writing && (
+            <Flex.Item shrink={1} height={TEXTAREA_INPUT_HEIGHT + 'px'}>
+              <Section
+                title="Insert Text"
+                fitted
+                fill
+                buttons={
+                  <Button.Confirm
+                    disabled={!savableData}
+                    content="Save"
+                    color="good"
+                    onClick={() => {
+                      if (textAreaText.length) {
+                        act('add_text', { text: textAreaText });
+                        setTextAreaText('');
+                      }
+                      if (Object.keys(inputFieldData).length) {
+                        act('fill_input_field', { field_data: inputFieldData });
+                        setInputFieldData({});
+                      }
+                    }}
+                  />
+                }>
+                <TextArea
+                  scrollbar
+                  noborder
+                  value={textAreaText}
+                  textColor={useColor}
+                  fontFamily={useFont}
+                  bold={useBold}
+                  height={'100%'}
+                  backgroundColor={paper_color}
+                  onInput={(e, text) => {
+                    setTextAreaText(text);
+
+                    if (this.scrollableRef.current) {
+                      let thisDistFromBottom =
+                        this.scrollableRef.current.scrollHeight -
+                        this.scrollableRef.current.scrollTop;
+                      this.scrollableRef.current.scrollTop +=
+                        thisDistFromBottom - this.lastDistanceFromBottom;
+                    }
+                  }}
+                />
+              </Section>
+            </Flex.Item>
+          )}
+        </Flex>
+      </>
+    );
+  }
+}
+
+/**
+ * @classdesc Real-time text preview section. When not editing, this is simply
```

Comment:
I think this tag is implied, so it's unnecessary.

Reply:
Removed.